scratchpad/tn_api/tn.py

```diff
@@ -98,7 +98,17 @@ def slice(self, slice_dict: dict) -> 'TensorNetwork':
                         slice_bounds.append(slice_dict[idx])
                     except KeyError:
                         slice_bounds.append(slice(None))
-                tn._tensors[current_tensor_ref] = tn._tensors[current_tensor_ref][tuple(slice_bounds)]
+                sliced_tensor = tn._tensors[current_tensor_ref][tuple(slice_bounds)]
+                tn._tensors[current_tensor_ref] = sliced_tensor
+        
+                # TODO: this is just a guess - but i am adding the ports from the popped edge back to the list of slices
```

Comment:
@danlkv where we left off last time is that calling einsum is failing on a sliced tn. it's likely because slicing the tn removes all edges - i don't think that behavior is intended, so my proposal for a fix is that in this section of logic (starting at line 76) when we pop an edge out, at some point we need to add it back.

i've added some pdb loops so that when we err out, we can repro it in the same run

here's an example error output
```
aa,ba->ab
[(4, 4, 2), (4, 4, 2)]
((Port(tensor_ref=0, ix=0), Port(tensor_ref=0, ix=1), Port(tensor_ref=1, ix=1)), (Port(tensor_ref=1, ix=0),))
2
operand has more dimensions than subscripts given in einstein sum, but no '...' ellipsis provided to broadcast the extra dimensions.
[1] > /app/scratchpad/tn_api/tn.py(172)contract()
-> keep_going = True
(Pdb++) c
Traceback (most recent call last):
  File "tn_api/tn.py", line 168, in contract
    return np.einsum(einsum_expr, *self._tensors)
  File "<__array_function__ internals>", line 200, in einsum
  File "/usr/local/lib/python3.8/dist-packages/numpy/core/einsumfunc.py", line 1371, in einsum
    return c_einsum(*operands, **kwargs)
ValueError: operand has more dimensions than subscripts given in einstein sum, but no '...' ellipsis provided to broadcast the extra dimensions.
```